packages/skill-lesson/trpc/routers/invoices.ts

```diff
@@ -18,47 +18,58 @@ export const invoicesRouter = router({
 
       const merchantCharge = await getMerchantCharge(merchantChargeId)
 
-      if (merchantCharge && merchantCharge.identifier) {
-        const charge = await stripe.charges.retrieve(
-          merchantCharge.identifier,
-          {
-            expand: ['customer'],
-          },
-        )
-
-        const purchase = await getPurchaseForStripeCharge(
-          merchantCharge.identifier,
-        )
-        const bulkCoupon = purchase && purchase.bulkCoupon
+      try {
+        if (merchantCharge && merchantCharge.identifier) {
+          const charge = await stripe.charges.retrieve(
+            merchantCharge.identifier,
+            {
+              expand: ['customer'],
+            },
+          )
 
-        let quantity = 1
-        if (purchase?.merchantSession?.identifier) {
-          const checkoutSession = await stripe.checkout.sessions.retrieve(
-            purchase?.merchantSession?.identifier,
-            {expand: ['line_items']},
+          const purchase = await getPurchaseForStripeCharge(
+            merchantCharge.identifier,
           )
+          const bulkCoupon = purchase && purchase.bulkCoupon
 
-          quantity = checkoutSession.line_items?.data[0].quantity || 1
-        } else if (bulkCoupon) {
-          quantity = bulkCoupon.maxUses
-        }
+          let quantity = 1
+          if (purchase?.merchantSession?.identifier) {
+            const checkoutSession = await stripe.checkout.sessions.retrieve(
+              purchase?.merchantSession?.identifier,
+              {expand: ['line_items']},
+            )
 
-        const product = await getProduct({
-          where: {id: purchase?.productId},
-        })
+            quantity = checkoutSession.line_items?.data[0].quantity || 1
+          } else if (bulkCoupon) {
+            quantity = bulkCoupon.maxUses
+          }
 
-        if (product && charge && purchase) {
-          return {
-            state: 'SUCCESS' as const,
-            result: {
-              product,
-              charge,
-              quantity,
-              bulkCoupon,
-              purchaseId: purchase?.id,
-            },
+          const product = await getProduct({
+            where: {id: purchase?.productId},
+          })
+
+          if (product && charge && purchase) {
+            return {
+              state: 'SUCCESS' as const,
+              result: {
+                product,
+                charge,
+                quantity,
+                bulkCoupon,
+                purchaseId: purchase?.id,
+                purchase,
+              },
+            }
           }
         }
+      } catch (error) {
+        console.error(error)
+        return {
+          state: 'FAILED' as const,
+          error:
+            'Unable to lookup the charge and related entities' +
+            (error as any).message,
```

Comment:
stripe call fails when charge doesn't exist which happens because the stripe charge is in egghead stripe 😭